File: py/IEC62325/MarketOperations/MarketSystem/MarketResults/MarketStatement.py
from IEC61968.Common.Document import Document
from IEC61970.Base.Domain.DateTime import DateTime


class MarketStatement(Document):
    """
    A statement is a roll up of statement line items. Each statement along with its line items provide the details of specific charges at any given time. Used by Billing and Settlement.
    @created 28-Dec-2023 8:23:22 PM
    """

    def __init__(self):
        # The end of a bill period.
        super().__init__()
        self.end = DateTime()

        # The version number of previous statement (in the case of true up).
        self.reference_number = ""

        # The start of a bill period.
        self.start = DateTime()

        # The date of which Settlement is run.
        self.trade_date = DateTime()

        # The date of which this statement is issued.
        self.transaction_date = DateTime()

File: java/IEC62325/MarketOperations/MarketSystem/MarketResults/MarketStatement.java
package IEC62325.MarketOperations.MarketSystem.MarketResults;

import IEC61970.Base.Domain.DateTime;
import IEC61970.Base.Domain.String;
import IEC61968.Common.Document;

/**
 * A statement is a roll up of statement line items. Each statement along with its
 * line items provide the details of specific charges at any given time.  Used by
 * Billing and Settlement.
 * @created 28-Dec-2023 8:23:22 PM
 */
public class MarketStatement extends Document {

	/**
	 * The end of a bill period.
	 */
	public DateTime end;
	/**
	 * The version number of previous statement (in the case of true up).
	 */
	public String referenceNumber;
	/**
	 * The start of a bill period.
	 */
	public DateTime start;
	/**
	 * The date of which Settlement is run.
	 */
	public DateTime tradeDate;
	/**
	 * The date of which this statement is issued. 
	 */
	public DateTime transactionDate;

	public MarketStatement(){

	}
}//end MarketStatement